File: scripts/inject-synergy-tags.js
const fs = require('fs');
const path = require('path');

const pagesDir = path.join(__dirname, '..', 'pages');

const widgetHtml = `            <!-- Community Synergy Tags -->
                <div class="relative inline-block w-full lg:w-auto" id="synergy-dock-wrap">
                    <button class="synergy-sidebar-btn group flex items-center justify-center gap-3 w-full lg:w-auto lg:pr-4 lg:py-1 lg:rounded-full hover:bg-slate-800 transition-colors" onclick="synTogglePopover(event)" type="button">
                        <div class="relative flex items-center justify-center w-10 h-10 lg:w-10 lg:h-10 rounded-full bg-[#1e293b] text-[#2dd4bf] group-hover:bg-[#2dd4bf] group-hover:text-[#0f172a] transition-all duration-300 shadow-md border border-indigo-500/30 lg:border-none">
                            <i class="fas fa-link text-lg"></i>
                            <span class="synergy-badge absolute -top-1 -right-1 bg-red-500 text-white text-[10px] font-bold px-1.5 py-0.5 rounded-full" style="display:none;"></span>
                        </div>
                        <span class="hidden lg:inline-block font-semibold text-gray-300 group-hover:text-white transition-colors">Synergies</span>
                    </button>
                    <div class="synergy-popover" id="synergy-popover">
                        <div class="synergy-popover-header">
                            <span class="synergy-popover-title">Synergies</span>
                            <span class="synergy-popover-close" onclick="synTogglePopover(event, false)">&#10005;</span>
                        </div>
                        <div class="tier-list" id="synergy-tier-list"></div>
                        <div class="search-wrap">
                            <div class="search-box">
                                <span class="icon">&#128269;</span>
                                <input id="synergy-search-input" type="text" placeholder="Search other archetypes&#8230;" oninput="onSynergySearch()" />
                            </div>
                            <div class="search-results" id="synergy-search-results"></div>
                            <div class="search-hint" id="synergy-search-hint">Top 3 shown &mdash; search for anything else.</div>
                        </div>
                    </div>
                </div>`;

let modifiedCount = 0;
let skippedCount = 0;

const files = fs.readdirSync(pagesDir).filter(f => f.endsWith('.html'));

for (const filename of files) {
    const filepath = path.join(pagesDir, filename);
    let content = fs.readFileSync(filepath, 'utf8');

    if (content.includes('synergy-dock-wrap')) {
        skippedCount++;
        continue;
    }

    const archetypeMatch = content.match(/renderDeckResourcesCompact\('deck-resources-compact',\s*'([^']+)'/);
    if (!archetypeMatch) {
        skippedCount++;
        continue;
    }
    const archetypeName = archetypeMatch[1];

    const target = '<div id="archetype-cards-browser" class="flex justify-center"></div>';
    const replacement = `${target}\n                <!-- Vertical Divider (Desktop) -->\n                <div class="hidden md:block w-px h-6 bg-indigo-500/30 mx-1"></div>\n${widgetHtml}`;

    if (content.includes(target)) {
        content = content.replace(target, replacement);
    } else {
        skippedCount++;
        continue;
    }

    if (!content.includes('assets/css/synergy-tags.css')) {
        const cssTarget = '<link rel="stylesheet" href="../assets/css/cookie-consent.css">';
        if (content.includes(cssTarget)) {
            content = content.replace(cssTarget, `<link rel="stylesheet" href="../assets/css/synergy-tags.css">\n    ${cssTarget}`);
        } else {
            content = content.replace('</head>', `    <link rel="stylesheet" href="../assets/css/synergy-tags.css">\n</head>`);
        }
    }

    if (!content.includes('assets/js/synergy-tags.js')) {
        const jsInject = '<script src="../assets/js/text-utils.js"></script>\n    <script src="../assets/js/synergy-tags.js"></script>';
        const jsTarget = '<script src="../assets/js/combo-system.js"></script>';
        if (content.includes(jsTarget)) {
            content = content.replace(jsTarget, `${jsInject}\n    ${jsTarget}`);
        } else {
            content = content.replace('</head>', `    ${jsInject}\n</head>`);
        }
    }

    const initScript = `initSynergyTags('${archetypeName}');`;
    if (!content.includes(initScript)) {
        const comboMatch = content.match(/(initCommunityCombos\([^)]+\);)/);
        if (comboMatch) {
            content = content.replace(comboMatch[1], `${comboMatch[1]}\n                ${initScript}`);
        } else {
            content = content.replace('</body>', `    <script>document.addEventListener('DOMContentLoaded', function () { ${initScript} });</script>\n</body>`);
        }
    }

    fs.writeFileSync(filepath, content, 'utf8');
    modifiedCount++;
    console.log(`Modified: ${filename}`);
}

console.log(`\n--- Summary ---`);
console.log(`Modified: ${modifiedCount}`);
console.log(`Skipped: ${skippedCount}`);

File: scripts/inject-synergy-tags.py
import os
import re

pages_dir = os.path.join(os.path.dirname(__file__), '..', 'pages')

widget_html = """            <!-- Community Synergy Tags: inside the resources dock -->
            <div class="relative inline-block" id="synergy-dock-wrap">
                <button class="synergy-dock-btn" onclick="synTogglePopover(event)" type="button">
                    <span class="synergy-badge" style="display:none;"></span>
                    <span>&#128279; Pairs Well With</span>
                    <span class="mini-icons" id="synergy-mini-icons"></span>
                </button>
                <div class="synergy-popover" id="synergy-popover">
                    <div class="synergy-popover-header">
                        <span class="synergy-popover-title">Pairs Well With</span>
                        <span class="synergy-popover-close" onclick="synTogglePopover(event, false)">&#10005;</span>
                    </div>
                    <div class="tier-list" id="synergy-tier-list"></div>
                    <div class="search-wrap">
                        <div class="search-box">
                            <span class="icon">&#128269;</span>
                            <input id="synergy-search-input" type="text" placeholder="Search other archetypes&#8230;" oninput="onSynergySearch()" />
                        </div>
                        <div class="search-results" id="synergy-search-results"></div>
                        <div class="search-hint" id="synergy-search-hint">Top 3 shown &#8212; search for anything else.</div>
                    </div>
                </div>
            </div>"""

modified_count = 0
skipped_count = 0

for filename in os.listdir(pages_dir):
    if not filename.endswith('.html'):
        continue
    filepath = os.path.join(pages_dir, filename)
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # 1. Skip files that shouldn't be touched or already have it
    if 'synergy-dock-wrap' in content:
        skipped_count += 1
        continue
        
    # Extract Archetype Name
    archetype_name = None
    m = re.search(r"renderDeckResourcesCompact\('deck-resources-compact',\s*'([^']+)'", content)
    if m:
        archetype_name = m.group(1)
        
    if not archetype_name:
        skipped_count += 1
        continue

    # 2. Inject into the dock
    # Find: <div id="archetype-cards-browser" class="flex justify-center"></div>
    # Then append vertical divider + widget
    target = '<div id="archetype-cards-browser" class="flex justify-center"></div>'
    replacement = f"""{target}
                <!-- Vertical Divider (Desktop) -->
                <div class="hidden md:block w-px h-6 bg-indigo-500/30 mx-1"></div>
{widget_html}"""
    
    if target in content:
        content = content.replace(target, replacement, 1)
    else:
        # If we can't find it, skip
        skipped_count += 1
        continue

    # 3. Inject CSS into <head>
    if 'assets/css/synergy-tags.css' not in content:
        # Put before </head> or cookie-consent.css
        if '<link rel="stylesheet" href="../assets/css/cookie-consent.css">' in content:
            content = content.replace('<link rel="stylesheet" href="../assets/css/cookie-consent.css">', 
                                      '<link rel="stylesheet" href="../assets/css/synergy-tags.css">\n    <link rel="stylesheet" href="../assets/css/cookie-consent.css">', 1)
        else:
            content = content.replace('</head>', '    <link rel="stylesheet" href="../assets/css/synergy-tags.css">\n</head>', 1)
            
    # 4. Inject JS into <head>
    if 'assets/js/synergy-tags.js' not in content:
        js_inject = '<script src="../assets/js/text-utils.js"></script>\n    <script src="../assets/js/synergy-tags.js"></script>'
        if '<script src="../assets/js/combo-system.js"></script>' in content:
            content = content.replace('<script src="../assets/js/combo-system.js"></script>', 
                                      f'{js_inject}\n    <script src="../assets/js/combo-system.js"></script>', 1)
        else:
            content = content.replace('</head>', f'    {js_inject}\n</head>', 1)
            
    # 5. Inject init script
    init_script = f"initSynergyTags('{archetype_name}');"
    if init_script not in content:
        # Try to find initCommunityCombos
        m_combo = re.search(r"(initCommunityCombos\([^)]+\);)", content)
        if m_combo:
            content = content.replace(m_combo.group(1), f"{m_combo.group(1)}\n                {init_script}", 1)
        else:
            content = content.replace('</body>', f"    <script>document.addEventListener('DOMContentLoaded', function () {{ {init_script} }});</script>\n</body>", 1)

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
    modified_count += 1
    print(f"Modified: {filename}")

print("\n--- Summary ---")
print(f"Modified: {modified_count}")
print(f"Skipped: {skipped_count}")
